plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -89,16 +96,23 @@ func NewMetricsReader(logger *zap.Logger, cfg config.Configuration) (*MetricsRea
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize prometheus client: %w", err)
 	}
+
 	mr := &MetricsReader{
 		client: promapi.NewAPI(client),
 		logger: logger,
+
+		metricsTranslator: dbmodel.New(cfg.OperationLabel),
+		callsMetricName:   buildFullCallsMetricName(cfg.MetricNamespace, cfg.CallsMetricName),
+		latencyMetricName: buildFullLatencyMetricName(cfg.MetricNamespace, cfg.LatencyMetricName, cfg.LatencyUnit),
+		operationLabel:    cfg.OperationLabel,
 	}
+
 	logger.Info("Prometheus reader initialized", zap.String("addr", cfg.ServerURL))
 	return mr, nil
 }
 
 // GetLatencies gets the latency metrics for the given set of latency query parameters.
-func (m *MetricsReader) GetLatencies(ctx context.Context, requestParams *metricsstore.LatenciesQueryParameters) (*metrics.MetricFamily, error) {
+func (m MetricsReader) GetLatencies(ctx context.Context, requestParams *metricsstore.LatenciesQueryParameters) (*metrics.MetricFamily, error) {
```

Comment:
MetricsReader's state doesn't change from external packages (and probably not at all) during its lifetime; so I thought to make it read-only.